crates/oxc_linter/src/rules/react/no_string_refs.rs

```diff
@@ -0,0 +1,277 @@
+                    });
+                  ",
+            None,
+        ),
+    ];
+
+    let fail = vec![
+        (
+            "
+              var Hello = createReactClass({
+                componentDidMount: function() {
+                  var component = this.refs.hello;
+                },
+                render: function() {
+                  return <div>Hello {this.props.name}</div>;
+                }
+              });
+            ",
+            None,
+        ),
+        (
+            "
+              var Hello = createReactClass({
+                render: function() {
+                  return <div ref=\"hello\">Hello {this.props.name}</div>;
+                }
+              });
+            ",
+            None,
+        ),
+        (
+            "
+              var Hello = createReactClass({
+                render: function() {
+                  return <div ref={'hello'}>Hello {this.props.name}</div>;
+                }
+              });
+            ",
+            None,
+        ),
+        (
+            "
+              var Hello = createReactClass({
+                componentDidMount: function() {
+                  var component = this.refs.hello;
+                },
+                render: function() {
+                  return <div ref=\"hello\">Hello {this.props.name}</div>;
+                }
+              });
+            ",
+            None,
+        ),
+        (
+            "
+              var Hello = createReactClass({
+                componentDidMount: function() {
+                var component = this.refs.hello;
+                },
+                render: function() {
+                  return <div ref={`hello`}>Hello {this.props.name}</div>;
+                }
+              });
+            ",
+            Some(serde_json::json!([{ "noTemplateLiterals": true }])),
+        ),
+        (
+            "
+              var Hello = createReactClass({
+                componentDidMount: function() {
+                var component = this.refs.hello;
+                },
+                render: function() {
+                  return <div ref={`hello${index}`}>Hello {this.props.name}</div>;
+                }
+              });
+            ",
+            Some(serde_json::json!([{ "noTemplateLiterals": true }])),
+        ),
+        (
+            "
+			        class Hello extends React.Component {
+			          componentDidMount() {
+                  var component = this.refs.hello;
+			          }
+			        }
+			      ",
+            None,
+        ),
+        (
+            "
+			        class Hello extends React.PureComponent {
+			          componentDidMount() {
+                  var component = this.refs.hello;
+			          }
+                render() {
+                  return <div ref={`hello${index}`}>Hello {this.props.name}</div>;
+                }
+			        }
+			      ",
+            Some(serde_json::json!([{ "noTemplateLiterals": true }])),
+        ),
```

Comment:
These cases are used for testing the es6 component. https://github.com/jsx-eslint/eslint-plugin-react/blob/master/tests/lib/rules/no-string-refs.js doesn't have these